src/redux/store.js

```diff
@@ -19,14 +20,35 @@ const initialState = {
   userPreferences: {
     filter: "ALL",
     sortBy: "NAME"
-  }
+  },
+  notifications: { }
 }
 
 const rootReducer = combineReducers({
   todoLists,
-  userPreferences
+  userPreferences,
+  notifications
 })
 
 const store = createStore(rootReducer, initialState, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
+setTimeout(() => {
+  Object.keys(todoListsState).forEach(todoListId => {
+    const todoList = todoListsState[todoListId];
+    const todos = todoList.todos;
+
+    Object.keys(todos).forEach(todoId => {
+      const todo = todos[todoId];
+      if (todo.active && todo.dueDate < new Date()) {
+        store.dispatch({
+          type: 'ADD_NOTIFICATION',
+          payload: {
+            todo
```

Comment:
You need to add the `todoListId` so that the notification in the state can store it and use it later to deactivate the todos.